src/plugins/data/server/search/search_service.ts

```diff
@@ -156,6 +157,17 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
       )
     );
 
+    // We don't want to register this because we don't want the client to be able to access this
+    // strategy, but we do want to expose it to other server-side plugins
+    // see x-pack/plugins/security_solution/server/search_strategy/timeline/index.ts
+    // for example use case
+    this.searchAsInternalUser = enhancedEsSearchStrategyProvider(
```

Comment:
@lukasolson now that we're exposing this here, should route logic be added to ensure routes throw based on space access? Right now only those that add RBAC explicitly will (which we are doing) hit logic that throws a 403, otherwise, call would go through.